packages/compiler/src/server/rename-file.ts

```diff
@@ -0,0 +1,64 @@
+import { Position, Range, TextEdit } from "vscode-languageserver";
+import { getSourceLocation } from "../core/diagnostics.js";
+import { getDirectoryPath, getRelativePathFromDirectory, resolvePath } from "../core/path-utils.js";
+import { ImportStatementNode } from "../core/types.js";
+
+/**
+ * Get the LSP TextEdit to update an import statment value.
+ * @param node Import statement node that should be updated
+ * @param newImport New import path
+ * @returns Lsp TextEdit
+ */
+export function getRenameImportEdit(node: ImportStatementNode, newImport: string): TextEdit {
+  const location = getSourceLocation(node);
+  const lineAndChar = location.file.getLineAndCharacterOfPosition(node.path.pos);
+  return TextEdit.replace(
+    Range.create(
+      Position.create(lineAndChar.line, lineAndChar.character + 1),
+      Position.create(lineAndChar.line, lineAndChar.character + node.path.value.length + 1),
+    ),
+    newImport,
+  );
+}
+
+export interface RenameFileParams {
+  oldPath: string;
+  newPath: string;
+  isDirRename: boolean;
+}
+
+export interface ReplaceImportResult {
+  newValue: string;
+  filePath: string;
+}
+
+/**
+ * Get the updated import value for a given rename operation.
+ * If the rename operation is not applicable, it returns undefined.
+ * @param target Import statement node that should be checked
+ * @param params Current rename operation parameters
+ * @returns Updated import value or undefined
+ */
+export function getUpdatedImportValue(
+  target: ImportStatementNode,
+  params: RenameFileParams,
+): ReplaceImportResult | undefined {
+  const filePath = getSourceLocation(target).file.path;
+  const fileDir = getDirectoryPath(filePath);
+  const importTarget = resolvePath(fileDir, target.path.value);
+  const oldPath = params.oldPath;
+  if (importTarget === oldPath || importTarget.startsWith(oldPath)) {
```

Comment:
the startsWith may be not accurate. for example:
the tsp originally contains an error for import 'c:\proj\sub-ab\module.tsp' which doesn't exist
we rename folder c:\proj\sub-a
then the startWith will return true for the c:\proj\sub-ab\module.tsp and change it unexpectedly.

Reply:
just tested with Typescript and it would rename the import even if they don't exists and I do feel like it makes sense anyway if you did actually enter that folder even if that file doesn't exists its most likely just a typo and renaming it would be more correct.